### DISCUSSION TITLE
Using SerialManager for a bleutooth serial device (was: Connect  SPC using rfcomm ?)

### DISCUSSION THREAD

**Discussion by anneranch** [blueman-project/blueman#2145] [Q&A] [answered]
I am trying to replace serial cable connection with Bluetooth.
I have working "blueman-manager " and I can "connect" using option "serial port". 
That gives me an option "rcfomm0"  and "send note".  All nice, but...
My remote device requires  specific serial format ( baud rate, bit size , stop bits etc.) 
HOW do I configure rfcomm0  as required.  I  knowrfcomm "does not care" but my device I am trying to communicate with does care.

**Comment by infirit** [comment, marked as answer]
AFAIK the same way you would for any serial device. For example `stty -F /dev/rfcommX cs8 9600` to set it up and then you can read/write directly from/to `/dev/rfcommX`.

Blueman can run a script for you, the script will be passed the following arguments `Address, Name, service name, uuid16s, rfcomm node`. Configure it in the plugin window (SerialManager entry)

**Reply by anneranch** [reply]
Thanks, looks good - unfortunately written in Python....
The SerialManager will work for my purpose just fine.

**Reply by anneranch** [reply]
I am little puzzled, I was under the impression that this is a "discussion", so why "mark as answer" ? 
I am using " SerialManager"  plugin and do not know what is "service name". Is it "SPP" ? Also how do I test the "options" entered ?
...and where is the "options" configuration stored -  I just lost what I entered while writing this ...
and do I have to reboot to run it?

**Comment by infirit** [comment]
> I am little puzzled, I was under the impression that this is a "discussion", so why "mark as answer" ?

You asked how to configure and use rfcomm as a serial device in the Q&A section. The answer was provided.

> I am using " SerialManager" plugin and do not know what is "service name". Is it "SPP" ? Also how do I test the "options" entered ? ...and where is the "options" configuration stored - 

The name is derived from the serial UUID, not sure what it is for serial devices.

> Also how do I test the "options" entered ? 

The script is run on connecting the bluetooth serial device but. What options do you need to test? 

> ...and where is the "options" configuration stored

Which options specifically are you referring to?

> I just lost what I entered while writing this ...

Blueman will handle connecting the device/service and creating the rfcomm device for you. From then on you should be able to use it like any other serial device. So do whatever you did with the previous setup.

**Comment by cschramm** [comment]
Seems like some fundamental misunderstanding.

The plugin setting is a path to a "script" - where "script" is actually any executable. There is no Python involved. You can provide a shell script, a binary or whatever executable. The executable gets passed those arguments in that order.

You can log inputs e.g. in a shell script like

```sh
#!/bin/sh

echo "$@" >/path/to/log_file
```

A script for @infirit's sample call could be

```sh
#!/bin/sh

stty -F "$5" cs8 9600
```

You might want to do that depending on the device, though, with the address in $1 being the best identifier.

**Comment by anneranch** [comment]
OK, you call it script, which in my understanding in Python terminology -
to execute "main" one calls Python script.
But that is not my main misunderstanding. If if use "serialmanager" plug in
and configure it - how do I execute such plug-in configuration / script?
There is an example how to "configure" , I assume rfcomm0 , and that is
what I like to verify.
PS I managed to clone the Python code, but having issues running "the
main"....

**Reply by cschramm** [reply]
You want to run blueman in some way to see what it's passing to the script? The main executable you might be looking for is `blueman-applet`.

Just log the arguments from inside the script instead. To find out the created device, just check the contents of `/dev`. `/dev/rfcomm0` is the best guess but it's dynamic and nothing that you should assume. That's why blueman passes you the correct path.

**Comment by infirit** [comment]
Unanswered and changed the title. Keep the discussion in here @anneranch .

**Comment by anneranch** [comment]
Thanks, that is what I was I looking for, unfortunately I forgot how to
save cloned code....
I have no issues running rfcomm or stty , I am working on adding my own
"response" code to the bluetooth manager.
But I have to have it running first....
